sdk/keyvault/Azure.Security.KeyVault.Keys/tests/DelayCreateKeyInterceptor.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using Azure.Core.Testing;
+using Castle.DynamicProxy;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Keys.Tests
+{
+    // TODO: Remove this implementation when https://github.com/Azure/azure-sdk-for-net/issues/8575 is fixed.
+    internal class DelayCreateKeyInterceptor : IInterceptor
+    {
+        private const string DelayEnvironmentVariableName = "AZURE_KEYVAULT_TEST_DELAY";
+        private const int DefaultDelay = 1000;
+
+        private readonly int _delay = DefaultDelay;
+        private readonly RecordedTestMode _mode;
+
+        internal DelayCreateKeyInterceptor(RecordedTestMode mode)
+        {
+            _mode = mode;
+
+            if (int.TryParse(Environment.GetEnvironmentVariable(DelayEnvironmentVariableName), out int delay))
+            {
+                if (delay >= 0)
+                {
+                    _delay = delay;
+                }
+            }
+        }
+
+        public void Intercept(IInvocation invocation)
+        {
+            if (_mode != RecordedTestMode.Playback && _delay > 0)
+            {
+                string name = invocation.Method.Name;
+                switch (name)
+                {
+                    case nameof(KeyClient.CreateEcKeyAsync):
+                    case nameof(KeyClient.CreateKeyAsync):
+                    case nameof(KeyClient.CreateRsaKeyAsync):
+                        TestContext.WriteLine("Delaying {0} by {1}ms", name, _delay);
+                        Thread.Sleep(_delay);
```

Comment:
Delaying a second for every call is sub-optimal, can we record the fact when last key creation happened and decide how long to delay based on that?

Reply:
We could store/compare that as a static, sure. Feel strongly it's worth it given this doesn't actually add all that much time to live/recording tests?

Reply:
Nope, not strongly. If id doesn't slow things too much I'm fine.

Reply:
It wouldn't be hard. Let me play around with it a bit.

Reply:
Change pushed. On my laptop while WFH, most delays were still the default max (1s) but a few came under in the mid- to high 900ms. I expect we'll see much better gains on the platform where 1 to 3 methods were failing per live run (linux).